cpp/src/arrow/compute/kernels/copy_data.h

```diff
@@ -0,0 +1,159 @@
+// Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
Can we name the header `copy_data_internal.h`? That way it won't get installed/exposed to outside users.